packages/kubernetes-1.15/kubelet-config

```diff
@@ -29,6 +29,20 @@ authorization:
 clusterDomain: {{settings.kubernetes.cluster-domain}}
 clusterDNS:
 - {{settings.kubernetes.cluster-dns-ip}}
+{{~#if settings.kubernetes.eviction-hard}}
+evictionHard:
+  {{~#each settings.kubernetes.eviction-hard}}
+  {{@key}}: "{{this}}"
+  {{~/each}}
+{{~/if}}
+kubeReserved:
```

Comment:
We defaults values (like `ephemeral-storage = 1Gi`) to `kubeReserved`, so kubeReserved gonna always be there. No need for `if`. 👍

Reply:
If we're defaulting `ephemeral-storage` to some value, we should document that in the README somehow. I think we should also still wrap kubeReserved in a "if" block like the other fields just in case since it doesn't hurt.

Reply:
Yeah. I'll document that, and I take this idea from EKS as well same as `cpu_reserved` and `memory_reserved`. right, it does not hurt, but I think since we default  `kubeReserved` in bottlerocket, `kubeReserved` will always have values there.
If it's unnecessary to wrap it in a `if` block?

Reply:
@etungsten If we're defaulting values for this - is it necessary to wrap in an "if" block?  I'm just trying to understand the reasons why we would

Reply:
It certainly isn't necessary. It just seems more consistent with the other fields that way. It also means the kubeconfig will still be valid even if we're somehow missing that setting for whatever reason.
But I'll relent and agree that if we know we're keeping that default for sure, we can just not have the if block. I'm ok with keeping it as is.

Reply:
My concern is that we are adding settings with our own defaults that were not there before? @etungsten is there any chance that these changes would result in different behavior for a user that hasn't set these settings (ephemeral-storage, for example)? I tried to figure out the answer but could not.

Reply:
The change would reserve some system resources by default for kubelet (just kubelet right now I think) (e.g. 1 gibibyte of ephemeral storage). Users shouldn't be seeing any behavioral changes unless they're squeezing the system to the limit so much so kubelet isn't operating normally. These values are aligned with what EKS does for their EKS-optimized AMIs so I think they make sense here for our EKS-specific K8S variants.